plugins/discourse-post-voting/plugin.rb

```diff
@@ -74,6 +74,22 @@ module ::PostVoting
 
   add_to_serializer(:user_card, :vote_count) { object.vote_count }
 
+  add_to_serializer(:topic_list_item, :topic_vote_count) do
+    return nil unless SiteSetting.post_voting_serialize_topic_votes_count
+    return nil unless object.is_post_voting?
+
+    if SiteSetting.post_voting_topic_vote_count_mode == "total"
+      DB.query_single(<<~SQL, topic_id: object.id).first || 0
+        SELECT SUM(qa_vote_count)
+        FROM posts
+        WHERE topic_id = :topic_id
+          AND deleted_at IS NULL
+      SQL
+    else
+      object.first_post&.qa_vote_count || 0
+    end
+  end
```

Comment:
How about this instead?

```suggestion
  add_to_serializer(:topic_list_item, :topic_vote_count) do
    if SiteSetting.post_voting_topic_vote_count_mode == "total"
      object.posts.sum(:qa_vote_count)
    else
      object.first_post&.qa_vote_count || 0
    end
  end

  add_to_serializer(:topic_list_item, :include_topic_vote_count?) do
    SiteSetting.post_voting_serialize_topic_votes_count && object.is_post_voting?
  end
```

Using AR to re-use the default scopes, and using the `:include_` "pattern" instead of returning `nil`.

Also, did you check for N+1s? 🤔